src/core_plugins/kbn_doc_views/public/views/table.js

```diff
@@ -13,16 +13,29 @@ docViewsRegistry.register(function () {
         hit: '=',
         indexPattern: '=',
         filter: '=',
-        columns: '='
+        columns: '=',
+        onAddColumn: '=',
+        onRemoveColumn: '=',
       },
       controller: function ($scope) {
         $scope.mapping = $scope.indexPattern.fields.byName;
         $scope.flattened = $scope.indexPattern.flattenHit($scope.hit);
         $scope.formatted = $scope.indexPattern.formatHit($scope.hit);
         $scope.fields = _.keys($scope.flattened).sort();
 
-        $scope.toggleColumn = function (fieldName) {
-          _.toggleInOut($scope.columns, fieldName);
+        $scope.canToggleColumn = function canToggleColumn() {
```

Comment:
Good point about the plural, thanks!

To get the precomputed value correct I would have to set up a watch (because `$scope.onAddColumn` and `$scope.onRemoveColumn` could change over time), which would probably already negate some of the benefit. I'll take a look at the profiler to see if it's worth it :wink: